bundles/org.openhab.binding.weathercompany/src/main/java/org/openhab/binding/weathercompany/internal/handler/WeatherCompanyHandler.java

```diff
@@ -0,0 +1,744 @@
+        refreshInterval = getRefreshIntervalSeconds();
+
+        // Construct the URL for querying the forecast
+        forecastUrl = buildForecastUrl();
+
+        // Construct the URL for querying the PWS observations
+        pwsUrl = buildPwsUrl();
+
+        // Schedule the job to refresh the forecast
+        scheduleRefreshJob();
+    }
+
+    @Override
+    public void dispose() {
+        cancelRefreshJob();
+        updateStatus(ThingStatus.OFFLINE);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // Handler doesn't handle any commands
+    }
+
+    /*
+     * Build the URL for requesting the 5-day forecast
+     */
+    private String buildForecastUrl() {
+        StringBuilder sb = new StringBuilder(BASE_FORECAST_URL);
+        // Set response type as JSON
+        sb.append("?format=json");
+        // Set language from config
+        sb.append("&language=").append(config.language);
+        // Set API key from config
+        sb.append("&apiKey=").append(config.apiKey);
+        // Set the units to Imperial or Metric
+        sb.append("&units=").append(getUnitsQueryString());
+        // Set the location from config
+        sb.append(getLocationQueryString());
+        String url = sb.toString();
+        logger.debug("Forecast URL is {}", url.replace(config.apiKey, REPLACE_API_KEY));
+        return url.toString();
+    }
+
+    private String getLocationQueryString() {
+        boolean validConfig = true;
+        StringBuilder sb = new StringBuilder();
+        switch (config.locationType) {
+            case CONFIG_LOCATION_TYPE_POSTAL_CODE:
+                if (StringUtils.trimToNull(config.postalCode) == null) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Postal code is not set");
+                    validConfig = false;
+                } else {
+                    sb.append("&postalKey=").append(config.postalCode.replace(" ", ""));
+                }
+                break;
+            case CONFIG_LOCATION_TYPE_GEOCODE:
+                if (StringUtils.trimToNull(config.geocode) == null) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Geocode is not set");
+                    validConfig = false;
+                } else {
+                    sb.append("&geocode=").append(config.geocode.replace(" ", ""));
+                }
+                break;
+            case CONFIG_LOCATION_TYPE_IATA_CODE:
+                if (StringUtils.trimToNull(config.iataCode) == null) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "IATA code is not set");
+                    validConfig = false;
+                } else {
+                    sb.append("&iataCode=").append(config.iataCode.replace(" ", "").toUpperCase());
+                }
+                break;
```

Comment:
`default`? not valid?

Reply:
Default is not valid. I added the default case.